### PR TITLE
Ensure all licences for wells are imported

### PR DESCRIPTION
**Pull request by stephenhillier**
Not just licences associated with an aquifer

Also:  Excel export now populates the licenced_status_code column from the nightly imported licences.

### CODE REVIEW

File: app/backend/wells/management/commands/export.py [hunk 1]
```diff
@@ -209,8 +210,18 @@ def generate_files(self, zip_filename, spreadsheet_filename):
...
  left join registries_organization on
  registries_organization.org_guid = well.org_of_person_responsible_guid
+
+-- WELL LICENCES: any well having at least 1 licence entry will be marked as Licensed.
+    left join (select well_tag_number, count(*) as cur_licences from well
+    join well_licences on
+    well.well_tag_number = well_licences.well_id
+    group by well_tag_number) as licence_q
+    on well.well_tag_number = licence_q.well_tag_number
+
```
Comment (dainetrinidad) [inline app/backend/wells/management/commands/export.py:220]: Maybe this comment can be moved out of this sql string? I like the intention, but it leaves the string vulnerable for modification (i.e. a typo or a change in formatting can invalidate it).

File: app/backend/wells/management/commands/export.py [hunk 2]
```diff
@@ -143,11 +143,12 @@ def generate_files(self, zip_filename, spreadsheet_filename):
         #######
         # WELL
         #######
-        well_sql = ("""select well_tag_number, identification_plate_number,
+        well_sql = ("""select well.well_tag_number, identification_plate_number,
```
Comment (dainetrinidad) [inline app/backend/wells/management/commands/export.py:146]: What was the reason behind specifying the table for this column in here? (Just curious)
This SQL could also be better if we left out ambiguous columns and specified the tables for all of them, even by using an alias `well w`...then `w.well_tag_number, w. identification_plate_number`. No need to change it all now, but once your SQL gets big enough to join multiple tables, it's more prone to bugs and invalid data.

File: app/backend/aquifers/management/commands/import_licences.py [hunk 3]
```diff
@@ -89,20 +89,24 @@ def process_row(self, row, use_dev_fixtures=False, well=None, aquifer=None):
...
             return
 
         logging.info("importing licence #{}".format(row['LICENCE_NUMBER']))
 
         if not well or not aquifer:
             # Check the Licence is for a valid Aquifer and Well
-            aquifer = Aquifer.objects.get(pk=row['SOURCE_NAME'])
-            well = Well.objects.get(pk=row['WELL_TAG_NUMBER'])
+            # the if check here allows this function to be called with a specific
+            # well or aquifer for dev/test environments.
```
Comment (dainetrinidad) [inline app/backend/aquifers/management/commands/import_licences.py:101]: The comment pertains to the "if" condition, which I'm guessing is above? Could we move that above the `if` statement to clean it up and provide less confusion?

**Review by dainetrinidad — APPROVED** [review]
@stephenhillier Approved! I don't see any big issues but I do have questions and suggestions to clean up some of the code.

File: app/backend/wells/management/commands/export.py (hunk 2 above)
Comment (stephenhillier) [reply app/backend/wells/management/commands/export.py:146]: the newly joined table `well_licences` also has a column `well_tag_number`, so I had to specify which table to select well_tag_number from.

**Pull request merged**